### PR TITLE
Feature/vethub 266

### PR DESCRIPTION
**Pull request by Jo-Zh**
implement SRS algorithm:
- to test: `yarn test`

### CODE REVIEW

File: tests/components/SrsSort.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// import { testMCQquestions } from "../testSeeds";
```
Comment (Mohamad11Dab) [inline tests/components/SrsSort.test.ts]: might want to delete this if not needed

File: src/utils/SrsSort.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// Function to transform a Question to CardInput
+const transformToCard = (card: MCQuestion): Card => ({
+  id: card._id.$oid,
+  last_review: card.lastAttempted as Date,
+  due: new Date(), // Default due date; you may adjust based on your logic
+  stability: 1.0, // initial value
+  difficulty: 2.5, // initial value
+  elapsed_days: Math.floor(
+    (new Date().getTime() - new Date(card.lastAttempted).getTime()) /
+      (1000 * 60 * 60 * 24),
```
Comment (Mohamad11Dab) [inline src/utils/SrsSort.ts]: Might want to avoid magic numbers

File: src/utils/SrsSort.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      reviewDue: updatedCard.reviewDue,
+    };
+  });
+
+// Sort questions based on the next review date
+export const sortedQuestions = (questions: MCQuestion[]) => {
+  return updatedQuestions(questions).sort(
+    (curr, next) => curr.reviewDue.getTime() - next.reviewDue.getTime(),
+  );
+};
```
Comment (Mohamad11Dab) [inline src/utils/SrsSort.ts:72]: The code currently updates all questions and then sorts them. This is fine, but can't performance become an issue with large data ? probably consider optimizing by updating and sorting at the same time ?

**Review by Mohamad11Dab — APPROVED** [review]
LGTM !

File: src/utils/SrsSort.ts (hunk 3 above)
Comment (Jo-Zh) [reply src/utils/SrsSort.ts:72]: The performance issue might be next ticket

File: tests/components/SrsSort.test.ts (hunk 1 above)
Comment (Jo-Zh) [reply tests/components/SrsSort.test.ts]: fixed

File: src/utils/SrsSort.ts (hunk 2 above)
Comment (Jo-Zh) [reply src/utils/SrsSort.ts]: added comments to explain

**Pull request merged**